plugins/zns/zns.c

```diff
@@ -462,8 +460,55 @@ static int finish_zone(int argc, char **argv, struct command *cmd, struct plugin
 static int open_zone(int argc, char **argv, struct command *cmd, struct plugin *plugin)
 {
 	const char *desc = "Open zones\n";
+	const char *zslba = "starting LBA of the zone for this command";
+	const char *zrwa = "Zone Random Write Area Allocation";
+	const char *select_all = "send command to all zones";
+	const char *timeout = "timeout value, in milliseconds";
+
+	int err, fd;
+
+	struct config {
+		__u64	zslba;
+		__u32	namespace_id;
+		bool    zrwa;
+		bool	select_all;
+		__u32	timeout;
+	};
+
+	struct config cfg = {
+	};
+
+	OPT_ARGS(opts) = {
+		OPT_UINT("namespace-id", 'n', &cfg.namespace_id,  namespace_id),
+		OPT_SUFFIX("start-lba",  's', &cfg.zslba,         zslba),
+		OPT_FLAG("zrwa",         'r', &cfg.zrwa,          zrwa),
+		OPT_FLAG("select-all",   'a', &cfg.select_all,    select_all),
+		OPT_UINT("timeout",      't', &cfg.timeout,       timeout),
+		OPT_END()
+	};
+
+	fd = parse_and_open(argc, argv, desc, opts);
+	if (fd < 0)
+		return errno;
+
+	if (!cfg.namespace_id) {
+		err = nvme_get_nsid(fd, &cfg.namespace_id);
+		if (err < 0) {
+			perror("get-namespace-id");
+			goto close_fd;
+		}
+	}
 
-	return zns_mgmt_send(argc, argv, cmd, plugin, desc, NVME_ZNS_ZSA_OPEN);
+	err = nvme_zns_mgmt_send(fd, cfg.namespace_id, cfg.zslba, NVME_ZNS_ZSA_OPEN,
+		cfg.select_all, cfg.zrwa, 0, NULL, cfg.timeout, NULL);
+	if (!err)
+		printf("zns-open-zone: Success zone slba:%"PRIx64" nsid:%d\n",
```

Comment:
The command already reported success - we don't need to produce what we already know from the arguments that were given.